statuslistener/statuslistener.go

```diff
@@ -0,0 +1,153 @@
+package statuslistener
+
+import (
+	"sort"
+	"time"
+
+	c "github.com/RedHatInsights/sources-api-go/config"
+	"github.com/RedHatInsights/sources-api-go/dao"
+	"github.com/RedHatInsights/sources-api-go/internal/events"
+	"github.com/RedHatInsights/sources-api-go/kafka"
+	logging "github.com/RedHatInsights/sources-api-go/logger"
+	m "github.com/RedHatInsights/sources-api-go/model"
+	"github.com/RedHatInsights/sources-api-go/util"
+)
+
+const (
+	SourcesStatusTopic      = "platform.sources.status"
+	GroupID                 = "sources-api-status-worker"
+	EventAvailabilityStatus = "availability_status"
+)
+
+var config = c.Get()
+
+type AvailabilityStatusListener struct {
+	*events.EventStreamProducer
+}
+
+func Run() {
+	avs := AvailabilityStatusListener{EventStreamProducer: NewEventStreamProducer()}
+	avs.subscribeToAvailabilityStatus()
+}
+
+func NewEventStreamProducer() *events.EventStreamProducer {
+	sender := &events.EventStreamSender{}
+	return &events.EventStreamProducer{Sender: sender}
+}
+
+func (avs *AvailabilityStatusListener) subscribeToAvailabilityStatus() {
+	if logging.Log == nil {
+		panic("logging is not initialized")
+	}
+
+	consumerConfig := kafka.ConsumerConfig{Topic: config.KafkaTopic(SourcesStatusTopic), GroupID: GroupID}
+	kafkaConfig := kafka.Config{KafkaBrokers: config.KafkaBrokers, ConsumerConfig: consumerConfig}
+
+	kf := &kafka.Manager{Config: kafkaConfig}
+
+	errorConsume := kf.Consume(avs.ConsumeStatusMessage)
+
+	if errorConsume != nil {
+		logging.Log.Errorf("Consumer kafka message error: %s", errorConsume.Error())
+	}
+}
+
+func (avs *AvailabilityStatusListener) ConsumeStatusMessage(message kafka.Message) {
+	var statusMessage StatusMessage
+	err := message.ParseTo(&statusMessage)
+	if err != nil {
+		logging.Log.Errorf("Error in parsing status message %s", err.Error())
+		return
+	}
+
+	et := avs.eventType(message)
+	if et != EventAvailabilityStatus {
+		return
+	}
+
+	logging.Log.Infof("Kafka message %s, %s received with payload: %s", message.Headers, message.Key, message.Value)
+
+	headers := avs.headersFrom(message)
+
+	avs.processEvent(statusMessage, headers)
+}
+
+func (avs *AvailabilityStatusListener) headersFrom(message kafka.Message) []kafka.Header {
+	if len(message.Headers) < 1 {
+		return []kafka.Header{}
+	}
+
+	headers := make([]kafka.Header, len(message.Headers))
+	for index, header := range message.Headers {
+		headers[index] = kafka.Header{Key: header.Key, Value: header.Value}
+	}
+
+	return headers
+}
+
+func (avs *AvailabilityStatusListener) eventType(message kafka.Message) string {
+	for _, header := range message.Headers {
```

Comment:
Probably can replace this with `m.GetHeader("event_type")`

Reply:
(function eventType removed)